apps/condo/domains/miniapp/components/Catalog/CardGrid.tsx

```diff
@@ -0,0 +1,188 @@
+  .tabs-hide-first {
+    .ant-tabs-tab:first-of-type {
+      display: none;
+    }
+    
+    .ant-tabs-tab:nth-of-type(2) {
+      margin-left: 0 !important;
+    }
+    .ant-tabs-nav-operations { display: none !important; }
+  }
+`
+
+export type TabContent = {
+    category: string
+    apps: Array<MiniAppOutput>
+}
+
+type TabPaneContentProps = {
+    tab: TabContent
+    fallback?: React.ReactElement
+}
+
+type CardGridProps = {
+    tabs: Array<TabContent>
+    search: string
+    resetSearch: () => void
+}
+
+const getCardsAmount = (width: number) => {
+    return Math.max(1, Math.floor(width / (MIN_CARD_WIDTH + CARD_GAP)))
+}
+
+const TabPaneContent: React.FC<TabPaneContentProps> = ({ tab, fallback }) => {
+    const intl = useIntl()
+    const NoAppsMessage = intl.formatMessage({ id: 'miniapps.catalog.noServicesInCategory' })
+    const [{ width }, refCallback] = useContainerSize<HTMLDivElement>()
+    const cardsPerRow = getCardsAmount(width)
+
+    if (!tab.apps.length) {
+        return fallback ? (
+            fallback
+        ) : (
+            (
+                <BasicEmptyListView
+                    image='dino/playing@2x.png'
+                >
+                    <Typography.Title level={3}>
+                        {NoAppsMessage}
+                    </Typography.Title>
+                </BasicEmptyListView>
+            )
+        )
+    }
+
+    return (
+        <Row gutter={CONTENT_SPACING} ref={refCallback}>
+            {tab.apps.map(app => (
+                <Col span={24 / cardsPerRow} key={`${cardsPerRow}:${app.id}`}>
+                    <AppCard
+                        id={app.id}
+                        type={app.type}
+                        connected={app.connected}
+                        name={app.name}
+                        description={app.shortDescription}
+                        logoUrl={app.logo}
+                        label={app.label}
+                    />
+                </Col>
+            ))}
+        </Row>
+    )
+}
+
+export const CardGrid: React.FC<CardGridProps> = ({ tabs, search, resetSearch }) => {
+    const intl = useIntl()
+    const NoAppsFoundMessage = intl.formatMessage({ id: 'miniapps.catalog.noServicesFound' })
+    const router = useRouter()
+    const { query: { tab } } = router
+
+    const [{ width }, setRef] = useContainerSize<HTMLElement>()
+    const sideTabs = width > TAB_SWITCH_THRESHOLD
+
+    const TabComponent = sideTabs ? SideBlockTabs : TopRowTabs
+    const categories = tabs.map(tab => tab.category)
+    const tabFromQuery = (tab && !Array.isArray(tab) && categories.includes(tab.toUpperCase())) ? tab.toUpperCase() : ALL_APPS_CATEGORY
+    const selectedTab = search ? SEARCH_TAB_KEY : tabFromQuery
+
+    const handleTabChange = useCallback((newKey) => {
+        const newRoute = `${router.route}?tab=${newKey}`
+        return router.push(newRoute).then(resetSearch)
+    }, [router, resetSearch])
+
+    const searchTab: TabContent = {
```

Comment:
It's useless here, since useMemo does not use deep-comparison under the hood.
And also we're pre-rendering rest of tabs as well, so why need to memoize just a single tab?)